### DISCUSSION TITLE
开启过量规则导致的相关问题

### DISCUSSION THREAD

**Discussion by xfljy** [gkd-kit/gkd#660] [General]
scene上看到的CPU占有率，原来是50%以下
![1.jpg](https://github.com/user-attachments/assets/f179486e-0c0f-4ae7-8b19-b72a79e968ee)

耗电（24h是更新后的，明显比以前多）
![2.jpg](https://github.com/user-attachments/assets/a6e9c617-b686-4730-8172-2a4fd2ddf627)

![3.jpg](https://github.com/user-attachments/assets/a57c1d02-848e-4e64-8ebd-9f3496da2d96)

关闭服务后CPU占用率依然不变
![4.jpg](https://github.com/user-attachments/assets/d0c0353c-3c8f-4e25-ac16-3ade4e81922b)

![5.jpg](https://github.com/user-attachments/assets/7afd4c94-2e65-4bf8-8294-9ef4d0097a0a)

日志为
https://f.gkd.li/16189954

**Comment by lisonge** [comment]
关于耗电的说明：<https://gkd.li/faq/#power>

---

另外你一个界面竟然有 79 个规则，你开启的规则太多了，而且一大堆重复的规则

![image](https://github.com/user-attachments/assets/418b2c05-5ec0-4f0f-ba62-a8deb8e41fef)

**Reply by xfljy** [reply]
我还想问一下，有没有可能这个会重复计算规则之类的。
以图中全局规则中的 全屏广告为例，我总共是三个订阅链接，其中有两个有全屏广告的规则，但是图中显示了6个，还有局部广告也只有两个订阅链接有，但是这里不知道显示了几个
![6.jpg](https://github.com/user-attachments/assets/b31b02ac-22c0-45bb-9ac8-d9888bd937e6)

原问题在重启之后得到了解决（或许是要清缓存之类的）
重启后的日志为https://f.gkd.li/16192020

最后还是要感谢大佬做出如此优质的APP

**Reply by lisonge** [reply]
因为用户界面的可操作最小规则单位是 group 规则组

日志里面输出的是 规则组 内部的所有子规则

如果一个组内有100个规则，那日志就显示100条，但是用户可操作的开关只有一个

**Comment by aoguai** [comment]
大量的重复规则存在肯定会耗电。很多订阅里的规则80%以上都是重复的，

如果你很有耐心，你可以自己一个个订阅里单独开启某个APP测试哪个最好的跳过效果，就开启哪个订阅中的哪个APP规则。

否则我建议你只用一个订阅即可。如果遇到跳不过，不是应该先考虑添加更多订阅，而是 反馈给对应的订阅作者 > 自己编写规则 > 添加订阅但只开启对应的规则

**Reply by popokcn** [reply]
对，最简单的是只开一个规则，然后有不完善的就自己写
